Add Cassandra and ScyllaDB docs (#17833)
gravitational/teleport: docs/pages/database-access/guides/cassandra-self-hosted.mdx

$ teleport db start \
  --token=/tmp/token \
  --auth-server=teleport.example.com:3080 \
  --name=example-cassandra \
  --protocol=cassandra \
  --uri=cassandra.example.com:9042 \
  --labels=env=dev
```

<Admonition type="note">

The `--auth-server` flag must point to the Teleport cluster's Proxy Service
endpoint because the Database Service always connects back to the cluster over a
reverse tunnel.

</Admonition>

</ScopedBlock>
<ScopedBlock scope={["cloud"]}>

Start the Teleport Database Service, pointing the `--auth-server` flag to the
address of your Teleport Cloud tenant:

```code
$ teleport db start \
  --token=/tmp/token \
  --auth-server=mytenant.teleport.sh:443 \
  --name=example-cassandra \
  --protocol=cassandra \
  --uri=cassandra.example.com:9042 \
  --labels=env=dev
```

</ScopedBlock>

<Admonition type="tip">
  You can start the Database Service using a configuration file instead of CLI flags.
  See the [YAML reference](../reference/configuration.mdx) for details.
</Admonition>

## Step 2/5. Create a Teleport user

(!docs/pages/includes/database-access/create-user.mdx!)

## Step 3/5. Set up mutual TLS

(!docs/pages/includes/database-access/tctl-auth-sign.mdx!)

> We will show you how to use `tctl auth sign` below.

Reviewer: ```suggestion
Follow the instructions below to generate TLS credentials for your database.
```
Since a user may not know yet what `tctl auth sign` does.

> When connecting to  Cassandra, sign the certificate for the hostname over

Reviewer: Should this paragraph be in the "Cassandra" tab below?

which Teleport will be connecting to it.

<Tabs>
  <TabItem label="Cassandra">
    ```code

>     $ auth sign --format=cassandra --host=cassandra.example.com --out=server --ttl=2190h

Reviewer: ```suggestion
    $ tctl auth sign --format=cassandra --host=cassandra.example.com --out=server --ttl=2190h
```

>     $ auth sign --format=cassandra --host=cassandra.example.com --out=server --ttl=2190h

Reviewer: ```suggestion
    $ tctl auth sign --format=cassandra --host=cassandra.example.com --out=server --ttl=2190h
```

>     $ auth sign --format=cassandra --host=cassandra.example.com --out=server --ttl=2190h

Reviewer: ```suggestion
    $ tctl auth sign --format=cassandra --host=cassandra.example.com --out=server --ttl=2190h
```

>     $ auth sign --format=cassandra --host=cassandra.example.com --out=server --ttl=2190h

Reviewer: ```suggestion
    $ tctl auth sign --format=cassandra --host=cassandra.example.com --out=server --ttl=2190h
```

>     $ auth sign --format=cassandra --host=cassandra.example.com --out=server --ttl=2190h

Reviewer: ```suggestion
    $ tctl auth sign --format=cassandra --host=cassandra.example.com --out=server --ttl=2190h
```

    ```

    (!docs/pages/includes/database-access/ttl-note.mdx!)

    The command will create two files:
    - `server.keystore` with  generated private key and user certificate in JKS format.
    - `server.truststore` with Teleport's certificate authority in JKS format.
  </TabItem>
  <TabItem label="Scylla">
    ```code
    $ tctl auth sign --format=scylla --host=scylla.example.com --out=server --ttl=2190h
    ```

    (!docs/pages/includes/database-access/ttl-note.mdx!)

    The command will create three files:
    - `server.cas` with Teleport's certificate authority
    - `server.key` with a generated private key
    - `server.crt` with a generated user certificate
  </TabItem>
</Tabs>

## Step 4/5. Configure Cassandra/Scylla

> <Tabs>

Reviewer: ```suggestion
Follow the instructions for your database to enable TLS communication with your Teleport cluster:

<Tabs>
```

> <Tabs>

Reviewer: ```suggestion
Follow the instructions for your database to enable TLS communication with your Teleport cluster:

<Tabs>
```

  <TabItem label="Cassandra">
    To configure Cassandra to accept TLS connections, add the following to your
    Cassandra configuration file, `cassandra.yaml`:
    ```conf
    client_encryption_options:
      enabled: true
      optional: false
      keystore: /path/to/server.keystore
      keystore_password: "password"
      require_client_auth: true
      truststore: /path/to/server.truststore
      truststore_password: "password"
      protocol: TLS
      algorithm: SunX509
      store_type: JKS
      cipher_suites: [TLS_RSA_WITH_AES_256_CBC_SHA]
    ```

>     Where the `password` is the value generated in the previous. step by `tctl auth sign` command.

Reviewer: ```suggestion
    In the configuration above, replace `"password"` with the value generated in the previous step by the `tctl auth sign` command.
```

  </TabItem>
  <TabItem label="Scylla">

>     To configure Cassandra to accept TLS connections, add the following to your

Reviewer: Should "Cassandra" be "ScyllaDB" here?

Author: Yes, It should thanks.

>     To configure Cassandra to accept TLS connections, add the following to your

Reviewer: Should "Cassandra" be "ScyllaDB" here?

## Step 3/5. Set up mutual TLS

(!docs/pages/includes/database-access/tctl-auth-sign.mdx!)

We will show you how to use `tctl auth sign` below.

When connecting to  Cassandra, sign the certificate for the hostname over
which Teleport will be connecting to it.

<Tabs>
  <TabItem label="Cassandra">
    ```code
    $ auth sign --format=cassandra --host=cassandra.example.com --out=server --ttl=2190h
    ```

    (!docs/pages/includes/database-access/ttl-note.mdx!)

    The command will create two files:
    - `server.keystore` with  generated private key and user certificate in JKS format.
    - `server.truststore` with Teleport's certificate authority in JKS format.
  </TabItem>
  <TabItem label="Scylla">
    ```code
    $ tctl auth sign --format=scylla --host=scylla.example.com --out=server --ttl=2190h
    ```

    (!docs/pages/includes/database-access/ttl-note.mdx!)

    The command will create three files:
    - `server.cas` with Teleport's certificate authority
    - `server.key` with a generated private key
    - `server.crt` with a generated user certificate
  </TabItem>
</Tabs>
For example, if your Cassandra server is accessible at `cassandra.example.com`,

> run:

Reviewer 2: Is this phrase complete?
